pyproject.toml

```diff
@@ -0,0 +1,129 @@
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+# Testing tools configuration
+[tool.coverage.run]
+branch = true
+
+[tool.coverage.report]
+show_missing = true
+
+[tool.pytest.ini_options]
+minversion = "6.0"
+log_cli_level = "INFO"
+markers = ["unstable"]
+
+# Formatting tools configuration
+[tool.black]
+line-length = 99
+target-version = ["py310"]
+
+[tool.isort]
+profile = "black"
+
+[tool.poetry]
+# name = "opensearch-dashboards-operator"
+name = "charm"
+version = "1.0"
+description = "opensearch-dashboards-operator"
+authors = []
+
+[tool.poetry.dependencies]
+python = "^3.10"
+tenacity = ">=8.0.1"
+pure-sasl = ">=0.6.2"
+cosl = ">=0.0.5"
+pydantic = "^1.10.7"
+pyyaml = "^6.0.1"
+poetry-plugin-export = ">=1.6"
+
+[tool.poetry.group.charm-libs.dependencies]
+ops = "^2.4.1"
+cryptography = "^39.0.0"
+
+[tool.poetry.group.fmt]
+optional = true
+
+[tool.poetry.group.fmt.dependencies]
+black = "^22.3.0"
+ruff = ">=0.1"
+isort = "^5.13.2"
+
+[tool.poetry.group.lint]
+optional = true
+
+[tool.poetry.group.lint.dependencies]
+ops = "^2.4.1"
+black = "^22.3.0"
+ruff = ">=0.1"
+codespell = ">=2.2.2"
+pyright = "^1.1.318"
+isort = "^5.12.0"
+flake8 = "^6.0.0"
+flake8-docstrings = "^1.7.0"
+flake8-copyright = "^0.2.4"
+flake8-builtins = "^2.1.0"
+pyproject-flake8 = "^6.0.0.post1"
+pep8-naming = "^0.13.3"
+
+[tool.poetry.group.unit]
+optional = true
+
+[tool.poetry.group.unit.dependencies]
+pytest = ">=7.2"
+coverage = {extras = ["toml"], version = ">7.0"}
+jsonschema = ">=4.10"
+pytest-mock = "^3.11.1"
+
+[tool.poetry.group.integration]
+optional = true
+
+[tool.poetry.group.integration.dependencies]
+ops = "^2.4.1"
+pytest = ">=7.2"
+juju = "^3.2.0"
+coverage = {extras = ["toml"], version = ">7.0"}
+pytest-operator = ">0.20"
+pytest-operator-cache = {git = "https://github.com/canonical/data-platform-workflows", tag = "v11.0.0", subdirectory = "python/pytest_plugins/pytest_operator_cache"}
```

Comment:
Not sure if that matters here, but you're using `v11.0.1` in your `ci.yaml`

Reply:
Done.